### DISCUSSION TITLE
Mode 3 suggestions

### DISCUSSION THREAD

**Discussion by englundjd** [Courseplay/courseplay#7594] [Ideas]
I have a couple suggestions that might be able to help with the overloader running into the combine, and make it run a little better.  

1. I know the overloader backs up when you get to the end of the field or when in a pocket after the combine gets done unloading, but if there is a way to get it to back up farther and guarantee that it will always back up (because sometimes it does not do it which causes an instant collision when using lands on the combine). This would also be beneficial in weird shaped fields where the headlands aren't straight. When I say farther I mean about twice as far as it currently goes, or make it so the user can set how far back it goes. I do have this issue with mode 2 also.

2.  Is there a way to make the combine wait just a little longer when pulling out of the pocket to get back into its lane while the cart backs out of the way? It seems like (especially with bigger equipment) things just don't get out of the way fast enough and the combine smacks into the side of the tractor.

3. When mode 3 is plotting a course is it possible to get it to always try to come in behind the combine? When it plots a course to go ahead of the combine it almost always does not get stopped right at the end of the course and ends up in the field which then creates a collision as the combine ends up to close and the tractor won't move. The other issue I have is that courseplay turns the overloader around to soon and the combine gets full before it reaches the cart.

4. The last thing I wanted to suggest is to put a timer on the overloader while unloading into the truck, once the timer runs out it forces the overloader to continue on the route. I have trouble where the truck takes off cause its full and the cart just hangs out at the dump point because it's not empty. Meanwhile the combine stops because its full and the cart is still waiting on the truck to get back to finish unloading. Also sometimes I run into issues where the cart empties completely and then just sits there. I think a timer of about a minute would work fine.

Let me know what you think and if you would like some more info.

**Comment by englundjd** [comment]
couple images of the overloader not backing up at the end of the field 
![20211031144638_1](https://user-images.githubusercontent.com/85418441/139600336-d7e0274a-c0cd-4e54-a0d0-8bad5004ad15.jpg)
![20211031144652_1](https://user-images.githubusercontent.com/85418441/139600340-f7ec1a05-15a5-4912-b635-83692e6825a5.jpg)

here the overloader is just sitting after filling the truck instead of leaving to go back and unload the combine one more time. 
![20211031150926_1](https://user-images.githubusercontent.com/85418441/139599746-c24026ff-c383-470a-ae1b-f4916fc85022.jpg)

These last 2 images are within 15 seconds of eachother, Courseplay recalculated the overloader route on the fly which did not end up helping the situation. I've both praised and cussed its ability to do that at different times and places.  

here the overloader would have been in the way had the combine not beat it there.
![20211031152300_1](https://user-images.githubusercontent.com/85418441/139600275-f2735bc7-5d58-473e-8726-0618d41dff00.jpg)

here the overloader pulled up from the front and miss judged when the combine would be full. Sometimes when it does this it will circle back around but usually it doesn't.
![20211031152346_1](https://user-images.githubusercontent.com/85418441/139600283-15edd295-f089-49bc-8302-7b29cbf27b29.jpg)

**Reply by KAmaia** [reply]
I am thumbs upping this for point number 3, I haven't used mode 3 yet, but in mode 2 it is definitely also an issue.

**Comment by rohne83** [comment]
For 1. this should not be a fixed setting. The code for serving the combines is the same for mode 3 (grain carts) and 2 (normal unloaders). For normal unloaders backing up twice as far as currently would create big driving issues for many european unloaders where turntable trailers are very common.
Think the issue in your first two screenshots is not the normal behaviour. Normally the grain cart should have backed up a bit in this situation as well to give space for the turning combine, but the cart just did not back up at all, right? Also have this every now and then, but that's not the normal behaviour but rather kind of a bug which has already been reported. See #7560 for reference (in German though). The screenshots which I added there later on might even be the same as one of the situations you wanted to describe with point 3?

For 4. with the grain cart not continuing: this works as designed. Timer had been requested last year already, but it was decided to not have it managed by a timer, but by fill levels. Whether the grain cart continues back to the field or keeps waiting for the next trailer is controlled by your setting for "drive on at less than". As the cart still has 59% fill level in your screenshot, you'd have to set this value to at least 60% for the grain cart to go back to the field.
Don't forget to set the pipe position as well (even for fixed pipes) as otherwise those comparisons might not work correctly in all cases.

**Reply by englundjd** [reply]
Thanks for the reply, I wasnt sure if those things had already been discussed or not. As far as the first 2 pictures, you are correct. It almost never backs up at the end of the field to allow the combine to turn and come through there without a collision when using the lands feature on the combine. 

As far as #7560, I have seen that issue pop up a few times recently while using the latest version of CP, If I roll back to the last version I do not see that issue (or it hasn't happened yet).
